src/core/public/chrome/chrome_service.tsx

```diff
@@ -98,13 +98,35 @@ export class ChromeService {
 
     const appTitle$ = new BehaviorSubject<string>('Kibana');
     const brand$ = new BehaviorSubject<ChromeBrand>({});
-    const isVisible$ = new BehaviorSubject(true);
     const isCollapsed$ = new BehaviorSubject(!!localStorage.getItem(IS_COLLAPSED_KEY));
     const applicationClasses$ = new BehaviorSubject<Set<string>>(new Set());
     const helpExtension$ = new BehaviorSubject<ChromeHelpExtension | undefined>(undefined);
     const breadcrumbs$ = new BehaviorSubject<ChromeBreadcrumb[]>([]);
     const badge$ = new BehaviorSubject<ChromeBadge | undefined>(undefined);
 
+    /**
+     * These observables allow consumers to toggle the chrome visibility via either:
+     *   1. Using setIsVisible() to trigger the next visibility$
+     *   2. Setting `chromeHidden` when registering an application, which will
+     *      reset the visibility whenever the next application is mounted
+     *   3. Having embed=true in the query string
+     */
+    const visibility$ = new BehaviorSubject(true);
+    const isVisible$ = merge(
+      visibility$,
+      application.currentAppId$.pipe(
+        map(
+          appId =>
+            !!appId &&
+            application.availableApps.has(appId) &&
+            !!application.availableApps.get(appId)!.chromeHidden
```

Comment:
The logic here seems backwards? Won't this default to `false` if `appId` is undefined or `chromeHidden` is undefined?